pkg/api/v1/clients/kube/controller/controller.go

```diff
@@ -187,9 +197,36 @@ func (c *Controller) enqueueSync(t eventType, old, new interface{}) {
 		runtime.HandleError(err)
 		return
 	}
+	// if no relevant changes were made, no reason to queue up work
+	if reflect.DeepEqual(old, new) {
+		return
+	}
+
+	oldYaml := SprintAny(old)
+	newYaml := SprintAny(new)
+	if oldYaml == newYaml {
+		return // reflect deep equals appears to work here, may want to remove this
+	}
+
+	// being more safe than we have to be.. remove??
+	// we should ensure event is namespaced to our watch namespaces!
+	if strings.Contains(key, "kube-system") {
```

Comment:
That's a very good idea! We could derive the namespaces we care about from the `namespaceWhitelist` on the `resourceClients` that we register with the `ResourceClientSharedInformerFactory`.

Reply:
i believe the informer registry is already namespaced, so i might have been able to get away with changing the `watchNamespaces` here